ops/pebble.py

```diff
@@ -751,34 +952,39 @@ def __repr__(self):
 class Check:
     """Represents a check in a Pebble configuration layer."""
 
-    def __init__(self, name: str, raw: typing.Dict = None):
+    def __init__(self, name: str, raw: Optional['_CheckDict'] = None):
         self.name = name
-        raw = raw or {}
-        self.override = raw.get('override', '')
+        dct = raw or {}  # type: _CheckDict
+        self.override = dct.get('override', '')  # type: str
         try:
-            self.level = CheckLevel(raw.get('level', ''))
+            level = CheckLevel(dct.get('level', ''))
         except ValueError:
-            self.level = raw.get('level')
-        self.period = raw.get('period', '')
-        self.timeout = raw.get('timeout', '')
-        self.threshold = raw.get('threshold')
-
-        http = raw.get('http')
+            level = dct.get('level')
+            # in to_dict we assume self.level is a CheckLevel instance,
+            # so we better make sure it is:
+            assert isinstance(level, CheckLevel), 'unable to ' \
+                                                  'determine level from %s' % level
```

Comment:
> in to_dict we assume self.level is a CheckLevel instance

Interesting, so it does (I probably wrote that). I think it does so incorrectly, though -- the intent is that `raw` is a JSON dict from the wire, so it won't have a `CheckLevel` in it. To be safe, I'd probably revert this `isinstance` to ensure we're not changing behaviour, and make `self.level` a `Union[CheckLevel, str]`.

Then in another PR (or even in this one, as long as we add a test), have `to_dict` support both str and CheckLevel for value, something like:

```python
'level': self.level.value if isinstance(self.level, CheckLevel) else self.level,
```

Reply:
Done! I also added a similar test for CheckInfo